src/main/java/org/sonarqube/auth/bitbucket/BitbucketIdentityProvider.java

```diff
@@ -103,6 +110,22 @@ public void callback(CallbackContext context) {
 
     GsonUser gsonUser = requestUser(scribe, accessToken);
     GsonEmails gsonEmails = requestEmails(scribe, accessToken);
+
+    String[] teamsRestriction = settings.teamRestriction();
+    if (teamsRestriction != null && teamsRestriction.length > 0) {
+
+      GsonTeams gsonTeams = requestTeams(scribe, accessToken);
+      if (gsonTeams == null || gsonTeams.getTeams() == null || gsonTeams.getTeams().size() == 0)
+        throw new UnauthorizedException(format("No teams found for current user: %s", gsonUser.getUsername()));
+
+      if (!gsonTeams.getTeams().stream().anyMatch(t -> Arrays.asList(teamsRestriction).contains(t.getUserName()))) {
+          LOGGER.trace(format("User %s is not part of restricted teams: '%s'", gsonUser.getUsername(),
```

Comment:
I'm not sure this log is really bringing value, as the message is returned to the user.